src/test/java/com/arpnetworking/metrics/common/sources/CollectdHttpSourceTest.java

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright 2016 Smartsheet
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.common.sources;
+
+import akka.actor.ActorRef;
+import akka.http.javadsl.model.HttpRequest;
+import akka.http.javadsl.model.HttpResponse;
+import akka.http.javadsl.model.headers.RawHeader;
+import com.arpnetworking.commons.observer.Observer;
+import com.arpnetworking.http.RequestReply;
+import com.arpnetworking.metrics.common.parsers.Parser;
+import com.arpnetworking.metrics.common.parsers.exceptions.ParsingException;
+import com.arpnetworking.metrics.mad.model.DefaultRecord;
+import com.arpnetworking.metrics.mad.model.Record;
+import com.arpnetworking.test.TestBeanFactory;
+import com.google.common.base.Charsets;
+import com.google.common.base.Throwables;
+import com.google.common.collect.Lists;
+import org.joda.time.DateTime;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Tests for the CollectdHttpSource.
+ *
+ * @author Brandon Arp (brandon dot arp at smartsheet dot com)
+ */
+public class CollectdHttpSourceTest extends ActorSourceTest {
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        _source = new CollectdHttpSource.Builder()
+                .setActorName("collectd")
+                .setName("collectd_source")
+                .setParser(_parser)
+                .build();
+        _source.attach(_observer);
+    }
+
+    @Test
+    public void testParsesEntity() throws ExecutionException, ParsingException {
+        final byte[] entity = "not a json document".getBytes(Charsets.UTF_8);
+        Mockito.when(_parser.parse(Mockito.any())).thenThrow(new ParsingException("test exception"));
+        dispatchRequest(HttpRequest.create().withEntity(entity));
+        Mockito.verify(_parser).parse(entity);
```

Comment:
This test seems incomplete. The input data doesn't matter since the parser always throws and nothing validates that when the parser does throw that the source behaves accordingly.